portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase

```diff
@@ -40,6 +40,209 @@ definition {
+				data = "<p>My content</p>",
+				ddmStructureId = "${ddmStructureId}",
+				externalReferenceCode = "erc",
+				label = "Text",
+				name = "Text",
+				title = "WC Second WebContent Title");
+		}
+
+		task ("Then I receive duplicate external reference code response") {
+			TestUtils.assertPartialEquals(
+				actual = "${response}",
+				expected = "Duplicate journal article external reference code erc");
+		}
+
+		task ("And Then another structured content with same erc is not being created") {
+			var response = HeadlessWebcontentAPI.filterStructuredContentInAssetLibrary(
+				assetLibraryId = "${depotId}",
+				filtervalue = "title%20eq%20%27WC%20Second%20WebContent%20Title%27");
+
+			HeadlessWebcontentAPI.assertProperNumberOfItems(
+				expectedTotalElement = "0",
+				responseToParse = "${response}");
+		}
+	}
+
+	@description = "Unable to get structured content by nonexistent assetLibraryId"
+	@priority = "4"
+	test CanReturnErrorWithGetStructuredContentInAssetLibraryByNonexistentAssetLibraryId {
+		property portal.acceptance = "true";
+
+		task ("When I make a GET request by non-existent assetLibraryID and a erc") {
+			var response = HeadlessWebcontentAPI.getStructuredContentsByErcInAssetLibrary(
+				assetLibraryId = "nonexistentId",
+				externalReferenceCode = "erc");
+		}
+
+		task ("Then I receive not found error response") {
+			TestUtils.assertPartialEquals(
+				actual = "${response}",
+				expected = "NOT_FOUND");
+		}
+
+		task ("And Then I receive error message in server console") {
+			AssertConsoleTextPresent(value1 = "Unable to get a valid asset library with ID nonexistentId");
+		}
+	}
+
+	@description = "Unable to get structured content by nonexistent erc"
+	@priority = "4"
+	test CanReturnErrorWithGetStructuredContentInAssetLibraryByNonexistentErc {
+		property portal.acceptance = "true";
+
+		task ("When I make a GET request by assetLibraryID and nonexistent erc") {
+			var depotId = JSONGroupAPI._getDepotIdByName(depotName = "Test Depot Name");
+
+			var response = HeadlessWebcontentAPI.getStructuredContentsByErcInAssetLibrary(
+				assetLibraryId = "${depotId}",
+				externalReferenceCode = "nonexistent-erc");
+		}
+
+		task ("Then I receive error message in response") {
+			TestUtils.assertPartialEquals(
+				actual = "${response}",
+				expected = "No JournalArticle exists with the key");
+		}
+	}
+
+	@description = "Structured content is updated by asset library id and erc"
+	@priority = "5"
+	test CanUpdateStructuredContentTitleInAssetLibraryByErc {
+		property portal.acceptance = "true";
+
+		task ("Given a structured content with a custom erc is created with a POST request in asset library") {
+			var depotId = JSONGroupAPI._getDepotIdByName(depotName = "Test Depot Name");
+			var ddmStructureId = WebContentStructures.getDdmStructureId(structureName = "content-structure");
+
+			var response = HeadlessWebcontentAPI.createStructuredContentInAssetLibrary(
+				assetLibraryId = "${depotId}",
```

Comment:
They are different, a depotID is the id of my asset library. And then a structured in asset library, after these provided, I can create a structured web content in asset library.

Reply:
so the id of assetLibrary can be of any other asset? or it has to be depot? what other examples of assetLibrary if available?

Reply:
more on the asset libraries [here](https://learn.liferay.com/dxp/latest/en/content-authoring-and-management/asset-libraries/creating-and-managing-asset-libraries.html) It may contain many different types of items 
![image](https://user-images.githubusercontent.com/35199951/178471347-0e8cb22b-1cdc-4fda-b3dc-4a94ef3855ea.png)
The macro to create it is called _getDepotIdByName but maybe for the sake of understanding this test we can call it `var assetLibrary` instead of `var depotId`?

Reply:
Thank you, updated to assetLibraryId.
![image](https://user-images.githubusercontent.com/1308942/178478967-8629792a-a42a-4cfc-969d-68fba335a6c6.png)